be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidator.scala

```diff
@@ -255,8 +257,23 @@ sealed class ElectionValidator(dbActorRef: => AskableActorRef) extends MessageDa
   }
 
   // not implemented since the back end does not receive a ResultElection message coming from the front end
+  // update : decentralization makes the back end receive a ResultElection message coming from GetMessagesById response from other servers.
+  // TODO: verify witness signatures. Problem : a [List[Signature]] does not tell us which witnesses produced this list of signatures.
   def validateResultElection(rpcMessage: JsonRpcRequest): GraphMessage = {
-    Left(PipelineError(ErrorCodes.SERVER_ERROR.id, "NOT IMPLEMENTED: ElectionValidator cannot handle ResultElection messages yet", rpcMessage.id))
+    def validationError(reason: String): PipelineError = super.validationError(reason, "KeyElection", rpcMessage.id)
+    rpcMessage.getParamsMessage match {
+      case Some(_: Message) =>
+        val (resultElection, laoId, senderPk, channel) = extractData[ResultElection](rpcMessage)
+        val electionId = channel.extractChildChannel
+        val electionQuestionResult = resultElection.questions
+        val witnessSignatures = resultElection.witness_signatures
+
+        runChecks(
+          checkNumberOfVotes(rpcMessage, channel, electionQuestionResult, dbActorRef, validationError(s"trying to send a ResultElection message with an invalid number of votes.\n"))
```

Comment:
I guess you could break that giant line into pieces, and I am not sure the `\n` is needed at the end.